### PR TITLE
Update Typings

### PR DESCRIPTION
**Pull request by brettguenther**
Fixes FENCE-1112

### CODE REVIEW

**Review by lmeier — APPROVED** [review]

File: src/definitions.ts [hunk 1]
```diff
@@ -8,11 +8,11 @@ export interface RadarPlugin {
   addListener(eventName: 'log', listenerFunc: (result: { message: string }) => void): Promise<PluginListenerHandle> & PluginListenerHandle;
   initialize(options: { publishableKey: string }): void;
   setLogLevel(options: { level: string }): void;
-  setUserId(options: { userId: string }): void;
+  setUserId(options: { userId?: string }): void;
```
Comment (lmeier) [inline src/definitions.ts:11]: Looks required in the `Plugin.swift` and `RadarPlugin.java` to me. Should switch the former to `let userId = call.getString("userId") ?? ""` if we want it optional.

But do we? It's not in `radar-sdk-ios`.

File: src/definitions.ts [hunk 2]
```diff
@@ -8,11 +8,11 @@ export interface RadarPlugin {
...
   getUserId(): Promise<object>,
-  setDescription(options: { description: string }): void;
+  setDescription(options: { description?: string }): void;
```
Comment (lmeier) [inline src/definitions.ts:13]: Same as `setUserId` comment

File: src/definitions.ts [hunk 3]
```diff
@@ -121,23 +121,31 @@ export interface Location {
...
+  courseAccuracy?: number;
 }
 
 export interface RadarUser {
   _id: string;
   userId?: string;
   deviceId?: string;
   description?: string;
+  stopped?: boolean;
+  segments?: RadarSegment[];
```
Comment (lmeier) [inline src/definitions.ts:138]: Should we add the rest? Glancing at a response:
```
debug?: boolean,
updatedAt: Date,
deviceId?: string,
installId?: string,
sdkVersion?: string,
deviceOS?: string,
deviceMake?: string,
deviceModel?: string,
userAgent?: string,
sessionId?: string,
location?: {
type: "Point",
coordinates: [lng, lat]
}
```

**Review by lmeier — COMMENTED** [review]
some questions re: optionals, and also suggestions for expanding some fields.

we should work towards having good typings for the entirety of our responses

File: src/definitions.ts (hunk 3 above)
Comment (brettguenther) [reply src/definitions.ts:138]: i dont think many of these are in the plugin mapping: https://github.com/radarlabs/radar-sdk-ios/blob/7769694225561ec697ccf8c9869a9b28d42171e0/RadarSDK/RadarUser.m#L298

Comment (brettguenther) [reply src/definitions.ts]: can keep to Point for now. we only seen to change to flatten the location in top level location objects.

**Pull request merged**